bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/converter/type/DateTimeTypeConverter.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.siemenshvac.internal.converter.type;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.ZonedDateTime;
+
+import org.openhab.binding.siemenshvac.internal.constants.SiemensHvacBindingConstants;
+import org.openhab.binding.siemenshvac.internal.converter.ConverterException;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.thing.type.ChannelType;
+import org.openhab.core.types.Type;
+
+import com.google.gson.JsonElement;
+
+/**
+ * Converts between a SiemensHvac datapoint value and an openHAB DecimalType.
+ *
+ * @author Laurent Arnal - Initial contribution
+ */
+public class DateTimeTypeConverter extends AbstractTypeConverter {
+
+    private final TimeZoneProvider timeZoneProvider;
+
+    public DateTimeTypeConverter(final TimeZoneProvider timeZoneProvider) {
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    protected boolean toBindingValidation(Type type) {
+        return true;
+    }
+
+    @Override
+    protected Object toBinding(Type type, ChannelType tp) throws ConverterException {
+        Object valUpdate = null;
+
+        if (type instanceof DateTimeType dateTime) {
+            valUpdate = dateTime.toString();
+        }
+
+        return valUpdate;
+    }
+
+    @Override
+    protected boolean fromBindingValidation(JsonElement value, String type) {
+        return true;
+    }
+
+    @Override
+    protected DateTimeType fromBinding(JsonElement value, String type, ChannelType tp) throws ConverterException {
+        if ("----".equals(value.getAsString())) {
+            return new DateTimeType();
+        } else {
+            String[] formats = { "EEEE, d. MMMM yyyy hh:mm", "d. MMMM yyyy hh:mm", "d. MMMM" };
+
+            for (int i = 0; i < formats.length; i++) {
+                try {
+                    SimpleDateFormat dtf = new SimpleDateFormat(formats[i]); // first example
+                    ZonedDateTime zdt = dtf.parse(value.getAsString()).toInstant()
+                            .atZone(this.timeZoneProvider.getTimeZone());
+
+                    if (i == 2) {
+                        zdt = zdt.withYear(2024);
+                    }
+
+                    return new DateTimeType(zdt);
+                } catch (ParseException ex) {
+                }
```

Comment:
change to:

 if (i == formats.length - 1) {
    throw new ConverterException("Can't parse the date for :" + value);
}

Reply:
Well I need to test all the possible formatter has I don't know before which one will feet the case.
If I catch outside the loop only, I willl only test the first formatter is there is a parsing exception with the first one.
Let me know if it make sense.

Reply:
It's not clear to me in which scenario the `for` loop will be exited normally without returning or throwing, i.e. how do you reach line 104?

Reply:
To clarify a little:

I one on the formatter in the formats tabs match the current value, you will exit on the return case In line 95. 
It will success, and the value will be corretly handle and convert to a DateTimeType.

If no formatter is working, you will exit on the throw clause on line 98.
It will be that we are not able to handle the value, for mainly two reason:
  1/ that we miss a formatter in the code, and that we will need to extend the binding to handle it.
  2/ that for some reason the value is corrupted, and so we can handle it.

And you have reason. I don't think that will never catch the line 104. But the compiler do not match that we will never be in this case, certainly because of the complexity of the loop, and the condition on the throw.
So we need to keep it just to prevent a compilation error because of missing return values !!!

Reply:
My point is that you could replace:
```java
                    if (i == formats.length - 1) {
                        throw new ConverterException("Can't parse the date for:" + value);
                    }
```

by either `continue;` or a comment explaining that we are proceeding to next format.

Then, as last line in the method:
```java
throw new ConverterException("Can't parse the date for:" + value);
```

replacing the unused logic:
```java
return new DateTimeType(ZonedDateTime.now(this.timeZoneProvider.getTimeZone()));
```

This should simplify the logic.

Reply:
Ok, so replace by something like this:

                    ZonedDateTime zdt = parsedDate.atZone(this.timeZoneProvider.getTimeZone());

                    return new DateTimeType(zdt);
                } catch (DateTimeParseException ex) {
                    // Silently ignore, we are proceeding to next format.
                }
            }
        }

        throw new ConverterException("Can't parse the date for:" + value);
    }

Reply:
Exactly, looks good. This way we have eliminated unused (thus confusing) logic, and we also got rid of mimmicking the exit condition for the loop, and instead will rely on the loop being exited naturally. I hope you agree it's easier to read and comprehend this way. Thanks!